src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -139,6 +146,30 @@ public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNu
         }
     }
 
+    public List<String> listProcess(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws IOException, InterruptedException {
+        LaunchResult result = launch(launchEnv, false, "top", containerId);
```

Comment:
`top` is even less likely to be there. What about `ps ax`? Or `pgrep`?

Reply:
this is [docker top](https://docs.docker.com/engine/reference/commandline/top/) command

Reply:
actual implementation for `docker top` (according to [this article](https://www.projectatomic.io/blog/2016/01/understanding-docker-top-and-ps/) ) is to run top on host but filter result to match container's processes.